### PR TITLE
CI: Add GitHub Action build jobs

### PR DESCRIPTION
**Pull request by spowelljr**

### CODE REVIEW

File: .github/workflows/linux.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+name: Build Linux
+on:
```
Comment (medyagh) [inline .github/workflows/linux.yml:2]: Lets make them run on Push to PR and Master as well

**Review by medyagh — CHANGES_REQUESTED** [review]
lets choose a more meaningful name for the name of the workflows, this will show up in the github action dasbhoard

so how about maybe

Build Linux
Build macOs
Build Windows

**Comment by spowelljr** [comment]
> lets choose a more meaningful name for the name of the workflows, this will show up in the github action dasbhoard
> 
> so how about maybe
> 
> Build Linux Build macOs Build Windows

The name of the workflow in the file is what shows up (`name: Build Windows`) and those are all named as you mentioned. The problem with using `build` in the file name is that the `.gitignore` has the following line `*build-*` so if I named the file `build-linux.yml` it would be ignored.

File: .github/workflows/linux.yml (hunk 1 above)
Comment (spowelljr) [reply .github/workflows/linux.yml:2]: Done

**Comment by medyagh** [comment]
> > lets choose a more meaningful name for the name of the workflows, this will show up in the github action dasbhoard
> > so how about maybe
> > Build Linux Build macOs Build Windows
> 
> The name of the workflow in the file is what shows up (`name: Build Windows`) and those are all named as you mentioned. The problem with using `build` in the file name is that the `.gitignore` has the following line `*build-*` so if I named the file `build-linux.yml` it would be ignored.

sounds good, in the future we are gonna have a Release Nightly job, so it will get a bit confusing to update them.

 I wonder if we should move all of them to one big workflow (run in parallel) or maybe name them PR but we can also think about it when we add those new jobs

**Pull request merged**